dlhub_sdk/client.py

```diff
@@ -450,25 +467,35 @@ def publish_servable(self, model):
             model.get_zip_file(zip_filename)
 
             # Get the authorization header token (string for the headers dict)
-            header = self.authorizer.get_authorization_header()
-
-            # Submit data to DLHub service
-            with open(zip_filename, 'rb') as zf:
-                reply = requests.post(
-                    slash_join(self.base_url, 'publish'),
-                    headers={"Authorization": header},
+            # header = self.authorizer.get_authorization_header()
+
+            # Use signed URL to upload zip file
+            SIGNED_URL_ENDPOINT = "https://api.dlhub.org/api/v1/publish/signed_url"
+            S3_DOWNLOAD_PREFIX = "https://dlhub-anl.s3.us-east-1.amazonaws.com/container_uploads/"
+            reply = requests.get(
+                SIGNED_URL_ENDPOINT)
+            signed_url = reply.json()
+            logger.debug(f'signed_url["url"] is {signed_url["url"]}')
+
+            with open('function.zip', 'rb') as zf:
+                http_response = requests.post(
+                    signed_url['url'],
+                    data=signed_url['fields'],
                     files={
-                        'json': ('dlhub.json', json.dumps(metadata), 'application/json'),
-                        'file': ('servable.zip', zf, 'application/octet-stream')
+                        'file': (signed_url['fields']['key'], zf, 'application/octet-stream')
                     }
                 )
+            if http_response.status_code != 200:
+                raise Exception(http_response.text)
+            metadata['dlhub']['transfer_method']['S3'] = S3_DOWNLOAD_PREFIX + signed_url['fields']['key']
+
+            # Ingest Model to DLHub
+            task = self._ingest(metadata)
 
             # Return the task id
-            if reply.status_code != 200:
-                raise Exception(reply.text)
-            return reply.json()['task_id']
+            return task['container_id']
```

Comment:
so I see you're actually returning the container id, just to double check, is that desired behavior?

Reply:
Great! TODO: resolve this in #container-service slack